hs/src/Reach/AST/SL.hs

```diff
@@ -161,15 +162,74 @@ data SLVal
   deriving (Eq, Generic)
 
 -- | Equivalence operation on flattened or simplified structures.
-class Equiv a where
+class (Eq a) => Equiv a where
   equiv :: a -> a -> Bool
+  default equiv :: (a -> a -> Bool)
+  equiv = (==)
+
+instance (Equiv a) => Equiv (Maybe a)
```

Comment:
Meaning that this says "If a supports Equiv, then Maybe a supports Equiv, by using == on the a instance (ignoring the equiv definition)?"